### PR TITLE
[FlagGems Operator Development Competition] feat: add asinh operator with high precision and stability

### PR DESCRIPTION
**Pull request by nicelynice**
## Description
Implemented `asinh`, `asinh_`, and `asinh_out` operators using Triton.

## Highlights
- **Numerical Stability**: Used $asinh(x) = \text{sign}(x) \cdot \ln(|x| + \sqrt{x^2 + 1})$ to handle negative values and large inputs accurately.
- **High Precision**: Max error is $2.38 \times 10^{-7}$, which is better than the requirement ($1.30 \times 10^{-6}$).
- **Performance**: Achieved ~0.99x speedup compared to native implementation on large shapes.

## Test Results
All tests passed across 1D to 5D shapes and special values.
[INFO] : FlagGems Unsupported GPU arch x86_64 specialization
```
================================================================================
Shape                     | Speedup    | Prec. (F32)  | Status    
--------------------------------------------------------------------------------
(1, 1)                    |      0.18x |     1.49e-08 | ✅ PASS    
(8, 8)                    |      0.18x |     1.19e-07 | ✅ PASS    
(64, 64)                  |      0.19x |     1.42e-07 | ✅ PASS    
(256, 256)                |      0.14x |     2.38e-07 | ✅ PASS    
(1024, 1024)              |      0.18x |     2.38e-07 | ✅ PASS    
(4096, 4096)              |      0.99x |     2.38e-07 | ✅ PASS    
(16, 3, 224, 224)         |      0.28x |     2.38e-07 | ✅ PASS    
(2, 4, 8, 16, 32)         |      0.21x |     2.38e-07 | ✅ PASS    
--------------------------------------------------------------------------------
Special Values (0, Inf, Large): ✅ PASS
Inplace Operation (asinh_):   ✅ PASS
```

### CODE REVIEW

**Comment by tengqm** [comment]
@nicelynice Please sign the CLA in order for your PR to be reviewed. Thanks.

File: src/flag_gems/ops/__init__.py [hunk 1]
```diff
@@ -117,6 +118,7 @@
 from flag_gems.ops.lerp import lerp_scalar, lerp_scalar_, lerp_tensor, lerp_tensor_
 from flag_gems.ops.linspace import linspace
 from flag_gems.ops.log import log
+from flag_gems.ops.log10 import log10, log10_, log10_out
```
Comment (tengqm) [inline src/flag_gems/ops/__init__.py]: These symbols are not exposed in `__all__`, why?

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -206,6 +209,7 @@ def torch_ge(v):
     ("linalg_vector_norm", vector_norm),
     ("linspace", linspace),
     ("log", log),
+    ("log10", log10),
```
Comment (tengqm) [inline src/flag_gems/__init__.py]: The other two are not exposed here, why?

File: src/flag_gems/ops/__init__.py (hunk 1 above)
Comment (tengqm) [reply src/flag_gems/ops/__init__.py]: This peculiar change seems out of the scope of this PR, am I understanding this right?

File: src/flag_gems/ops/asinh.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+def asinh(input):
+    return asinh_kernel(input)
+
+def asinh_out(input, out):
+    asinh_kernel(input, out0=out)
+    return out
+
+def asinh_(input):
+    asinh_kernel(input, out0=input)
+    return input
```
Comment (tengqm) [inline src/flag_gems/ops/asinh.py]: I believe we need some test cases for these operators.

File: tests/test_asinh.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
+import torch
+import pytest
+import numpy as np
+from flag_gems.ops.asinh import asinh, asinh_, asinh_out
+
+# ==============================================================================
+# 1. 严格遵循 4.1.1 精度评价标准定义
```
Comment (tengqm) [inline tests/test_asinh.py]: Please write comments in English.

File: tests/test_asinh.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
+import torch
+import pytest
+import numpy as np
+from flag_gems.ops.asinh import asinh, asinh_, asinh_out
```
Comment (tengqm) [inline tests/test_asinh.py]: The import grouping can be improved.

File: tests/test_asinh.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    )
+    ref = torch.asinh(special_cases)
+    res = asinh(special_cases)
+    
+    torch.testing.assert_close(res, ref, equal_nan=True)
+
+# ==============================================================================
+# 5. 性能与精度对比工具 (用于生成提交要求的完整对比表)
+# ==============================================================================
+def test_asinh_performance_table():
```
Comment (tengqm) [inline tests/test_asinh.py]: We typically don't mix accuracy tests with performance tests in the same file.
Please check some other test cases for reference.

File: tests/test_asinh.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+        t_gems = benchmark_op(asinh, x)
+        speedup = t_torch / t_gems
+        
+        # 3. 打印行
+        dtype_str = str(dtype).split('.')[-1]
+        print(f"{str(shape):<25} | {dtype_str:<10} | {t_torch:>10.4f}   | {t_gems:>10.4f}   | {speedup:>8.2f}x  | {max_diff:.2e}")
+    
+    print("="*105 + "\n")
+
+if __name__ == "__main__":
```
Comment (tengqm) [inline tests/test_asinh.py]: We use `pytest` to drive unit tests, so there is no need to have a main function.

File: src/flag_gems/ops/asinh.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import triton
+import triton.language as tl
+
+from flag_gems.utils import pointwise_dynamic
+
+
+@pointwise_dynamic(promotion_methods=[(0, "INT_TO_FLOAT")])
+@triton.jit
+def asinh_kernel(x):
+    # 数学公式: asinh(x) = ln(x + sqrt(x^2 + 1))
```
Comment (tengqm) [inline src/flag_gems/ops/asinh.py]: Please avoid writing Chinese in code (including comments).

File: benchmark/test_asinh_perf.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
+import time
+
+import torch
+from tabulate import tabulate
+
+import flag_gems
+
+
+def benchmark_asinh(shape, dtype, iters=100):
+    # 初始化数据
```
Comment (tengqm) [inline benchmark/test_asinh_perf.py]: There are other cases where comments should be written in English, please double check. Thanks.

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
